dashboard/app/controllers/home_controller.rb

```diff
@@ -1,13 +1,19 @@
 class HomeController < ApplicationController
   before_filter :authenticate_user!, only: :gallery_activities
 
-
   # Don't require an authenticity token on set_locale because we post to that
   # action from publicly cached page without a valid token. The worst case impact
   # is that an attacker could change a user's language if they fooled them into
   # clicking on a link.
   skip_before_action :verify_authenticity_token, :only => 'set_locale'
 
+  # Sets the client state key for the current user and redirects to params[:redirect] or
+  # the root if unspecified.
+  def update_login
```

Comment:
Note that the user does briefly see the URL in the address bar (unless we use a different string in the URL) so I didn't want to do something too frightening